airflow/metrics/statsd_logger.py

```diff
@@ -51,26 +51,22 @@ def prepare_metric_name_with_tags(fn: T) -> T:
 
     @wraps(fn)
     def wrapper(
-        self, metric_name: str | None = None, tags: dict[str, str | int] | None = None
+        self, metric_name: str | None = None, tags: dict[str, str] | None = None
```

Comment:
If you are no longer accepting `int` as a value type here, then I don't think you need to do the casting on L60?   I'm, not sure why you had added `int` as a valid type and then removed it thought, so one or the other should be fixed.

If `tags` is going to be a `dict[str, str]` then there is no need to cast it to a `dict[str, str]` on L60.  If it is going to be declared as `dict[str, str | int]` then we should keep the casing.

Reply:
We've gone back and forth on this one a couple times.  Can you confirm that it makes sense how it is now, or did I confuse you and you are just appeasing me?  If you need to chat about it, hit me up on Slack and we can talk it through.